Source/DafnyCore/Resolver/ModuleResolver.cs

```diff
@@ -1130,12 +1130,14 @@ public partial class ModuleResolver {
       // ---------------------------------- Pass 1 ----------------------------------
       // This pass does the following:
       // * desugar functions used in reads clauses
+      // * fills in "reads *" clauses on methods, when --reads-clauses-on-methods is used
```

Comment:
Just to confirm, do you think is this the "right" place to set defaults in general? From the original PR, we agreed it might have been cleaner to move in the other direction, to immediately fill in the default during parsing.

Reply:
(Responding a bit late.) If a default can be determined already in the parser, that may be a good place. At the other extreme, if the default matters only for verification (which I think is the case here), then we could choose never to update the AST with the default, and instead just take the appropriate action in the verifier. Here, it's now done in "Pass 1", which is alright. The one place we should avoid placing it, I think, is in the name resolution + type inference ("Pass 0"), because that pass is quite delicate.